src/elm/interval.coffee

```diff
@@ -235,18 +235,18 @@ module.exports.Collapse = class Collapse extends Expression
         if typeof b.low?.durationBetween == 'function'
           # handle DateTimes using durationBetween
           if a.high?.after b.low # overlap
-            a.high = b.high if !b.high? or b.high.after a.high
+            a.high = b.high if (!b.high? and b.low.before a.high) or b.high.after a.high
           else if a.high?.durationBetween(b.low, perWidth.unit).high <= perWidth.value
             a.high = b.high
-          else if a.high? and !b.high?
+          else if (a.high? and !b.high?) and (b.low.before a.high)
             a.high = b.high
           else
             collapsedIntervals.push a
             a = b
         else if typeof b.low?.sameOrBefore == 'function'
           if a.high? and b.low.sameOrBefore doAddition(a.high, perWidth)
             a.high = b.high if !b.high? or b.high.after a.high
-          else if a.high? and !b.high?
+          else if a.high? and !b.high? and b.low.sameOrBefore doAddition(a.high, perWidth)
```

Comment:
I think this is another case of an `else if` statement that will never be true.  The only difference between this and line 247 is that you also check for `!b.high` here -- that's _more_ restrictive than the initial `if` statement -- so nothing would ever fail the initial `if` and pass this `else if`.  I think you can safely delete the `else if`.